GeoQuiz/app/src/main/java/com/example/geoquiz/QuizViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.geoquiz
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+
+private const val TAG="MainActivity"
+
+class QuizViewModel : ViewModel() {
+
+     var mCurrentIndex: Int=0
+
+     var mQuestionBank= listOf<Question>(
+        Question(R.string.question_australia,true),
+        Question(R.string.question_oceans, true),
+        Question(R.string.question_mideast, false),
+        Question(R.string.question_africa, false),
+
+        )
+
+    val currentQuestionAnswer: Boolean
+    get()=mQuestionBank[mCurrentIndex].answerTrue
+
+    val currentQuestionText: Int
+    get()=mQuestionBank[mCurrentIndex].textResId
+
+    fun moveToNext(){
+        mCurrentIndex=(mCurrentIndex+1)%mQuestionBank.size
+    }
+
+    fun moveToPrev(){
+        mCurrentIndex=(mCurrentIndex-1)%mQuestionBank.size
+        if(mCurrentIndex<0){
+            mCurrentIndex=mQuestionBank.size-1
+        }
+    }
+
+   /* init{
```

Comment:
Commented code can be deleted. The remote branch will remember this code anyway so in case you need it at some point in the future you can always checkout this branch